app_legacy/Helpers/database/player-rank.php

```diff
@@ -59,6 +59,33 @@ function recalculatePlayerPoints(string $user): bool
     return (bool) $dbResult;
 }
 
+function countUserAchievements(?string $user, ?array &$dataOut): bool
+{
+    if (empty($user) || !isValidUsername($user)) {
+        return false;
+    }
+
+    $query = "SELECT User, 
+                SUM(CASE WHEN aw.HardcoreMode = " . UnlockMode::Hardcore . " THEN 1 ELSE 0 END) AS RAAchievements,
+                SUM(CASE WHEN aw.HardcoreMode = " . UnlockMode::Softcore . " THEN 1 ELSE 0 END) AS TotalAchievements
+                FROM Awarded AS aw
+                LEFT JOIN Achievements AS ach ON ach.ID = aw.AchievementID
+                LEFT JOIN GameData AS gd ON gd.ID = ach.GameID
+                WHERE aw.User =:username 
+                AND ach.Flags = " . AchievementType::OfficialCore . "
+                AND gd.ConsoleID NOT IN (100, 101)";
```

Comment:
The first time I tried to load Maxmilyn's page on my dev server, this query took 2.4 seconds. Repeated refreshing dropped it to ~300ms.

Can this feature wait for the denormalized data? If an absolute total isn't available, it should still be faster to sum per-game totals.

Reply:
Oh my I didn't realize how slow it was on the initial run, this led me to realize there was a much simpler way to just increment the count in the existing loop through the game data pulled for the completion widget so it shouldn't impact performance at all now